### PR TITLE
[FX-4584] Migrate Drawer to TailwindCSS

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-4584]

### Description

Migrate Drawer to TailwindCSS. Based on the modal PR as it shares some code.
- tested in [SP](https://github.com/toptal/staff-portal/pull/12331) and [CP](https://github.com/toptal/client-portal/pull/8387)

### How to test

- [Temploy](https://picasso.toptal.net/fx-4584-drawer?path=/story/components-drawer--drawer)
- compare [Temploy](https://picasso.toptal.net/fx-4584-drawer?path=/story/components-drawer--drawer) with the [master storybook](https://picasso.toptal.net/?path=/story/components-drawer--drawer)
- test [Drawer](https://t-client-portal-pr-8387-storybook.toptal.net/?path=/story/screeninginsightsdrawer--default) in the CP storybook
- 

### Screenshots

CP 
image

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4584]: https://toptal-core.atlassian.net/browse/FX-4584?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: packages/base/Drawer/src/Drawer/Drawer.tsx [hunk 1]
```diff
@@ -83,35 +96,54 @@ export const Drawer = (props: Props) => {
...
-      {...rest}
+    <Modal
       open={open}
+      ref={ref}
+      className={cx(className, 'fixed z-drawer inset-0')}
+      slots={{
+        backdrop: Backdrop,
+      }}
+      slotProps={{
+        // @ts-ignore
```
Comment (augustobmoura) [inline packages/base/Drawer/src/Drawer/Drawer.tsx:107]: Maybe we should explain this `@ts-ignore`, wdyt?
Can you avoid this any other way? Maybe using an assertion or type guard. Why is `invisible` not a valid property anyway?

Comment (TomasSlama) [reply packages/base/Drawer/src/Drawer/Drawer.tsx:107]: fixed, thank you for noticing, it was left over

File: .changeset/itchy-adults-film.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso-slide': major
+---
+
+- initial release of the component
```
Comment (sashuk) [inline .changeset/itchy-adults-film.md:5]: Screenshot 2024-03-22 at 10 34 38

😄

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: cypress/component/Drawer.spec.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ const DrawerExample = (
       <Button data-testid='trigger' onClick={onOpen}>
         Show drawer
       </Button>
-      <Drawer {...props}>
+      <Drawer onClose={() => {}} {...props}>
```
Comment (sashuk) [inline cypress/component/Drawer.spec.tsx:25]: Can we have a default value in `<Drawer/>` for `onClose` instead?

Comment (TomasSlama) [reply cypress/component/Drawer.spec.tsx:25]: Good catch, I forgot to remove these. I added them when I didn't understand why I have changes in my screenshots, those will be fixed when I rebase on https://github.com/toptal/picasso/pull/4204. So I will remove these changed in spec files

Comment (sashuk) [reply cypress/component/Drawer.spec.tsx:25]: @TomasSlama Got it, thank you!

**Review by sashuk — APPROVED** [review]
Tested in Staff Portal and compared in Storybook in a responsive manner. Code looks good!

**Review by mkrl — APPROVED** [review]
:rocket:

**Review by augustobmoura — APPROVED** [review]

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Pull request merged**